syntax = "proto3";

package scanner.proto;

message NetDescriptor {
  string model_path = 1;
  string model_weights_path = 2;

  repeated string input_layer_names = 3;
  repeated string output_layer_names = 4;

  int32 input_width = 5;
  int32 input_height = 6;

  repeated float mean_colors = 7;
  // or
  repeated float mean_image = 8;
  int32 mean_width = 9;
  int32 mean_height = 10;

  bool normalize = 11;
  bool preserve_aspect_ratio = 12;
  bool transpose = 13;
  int32 pad_mod = 14;
  bool uses_python = 15;
}

message CaffeInputArgs {
  NetDescriptor net_descriptor = 1;
  int32 batch_size = 2;
}

message CaffeArgs {
  NetDescriptor net_descriptor = 1;
  int32 batch_size = 2;
}

message FacenetArgs {
  CaffeArgs caffe_args = 1;
  string templates_path = 2;
  float scale = 3;
  float threshold = 4;
}

message CPM2Args {
  CaffeArgs caffe_args = 1;
  float scale = 2;
}

message OpenPoseArgs {
  string model_directory = 1;

  int32 pose_num_scales = 2;
  float pose_scale_gap = 3;

  bool compute_hands = 4;
  int32 hand_num_scales = 5;
  float hand_scale_gap = 6;

  bool compute_face = 7;
}